### PR TITLE
openamp sample integration/app

### PR DESCRIPTION
**Pull request by galak**
Example of using open-amp.

This example docs need some cleaning up, but this is based on #7415 and #7161

### CODE REVIEW

File: boards/arm/lpcxpresso54114_m0/doc/lpcxpresso54114.rst [hunk 1]
```diff
@@ -0,0 +1,61 @@
+.. _lpcxpresso54114_m0:
+
+NXP LPCXPRESSO54114
```
Comment (dbkinder) [inline boards/arm/lpcxpresso54114_m0/doc/lpcxpresso54114.rst]: might-should add ``M0`` or maybe ``(M0 core)`` to this title.  Without it, it has the same title as the board doc we've already got.

File: boards/arm/lpcxpresso54114_m0/doc/lpcxpresso54114.rst [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+tools are supported for this board in Zephyr, therefore you should use the
+Segger J-Link firmware variant.
+
+Before you start using Zephyr on the LPCXpresso54114, download and run
+`LPCScrypt`_ to update the LPC-Link2 firmware to the latest version, currently
+``Firmware_JLink_LPC-Link2_20160923.bin``. Serial communication problems, such
+as dropping characters, have been observed with older versions of the firmware.
+
+The code for the secondary core is linked into the primary core binary file.
+Startup code copies the secondary core's code into an appropriate location
```
Comment (dbkinder) [inline boards/arm/lpcxpresso54114_m0/doc/lpcxpresso54114.rst]: Might be work clarifying (assuming I've got this right):

    Startup code for the primary core copies the secondary core's code into an
    appropriate location in RAM and starts the secondary core's execution.

**Review by dbkinder — DISMISSED** [review]

**Comment by galak** [comment]
@dbkinder can you review this now from a doc perspective.

**Comment by galak** [comment]
This is the first step towards a fuller solution for #3065.  We expect in the future to update the openamp implementation both for improved integration with Zephyr, as well as reduce code footprint from an improved openamp library.

**Review by SebastianBoe — DISMISSED** [review]
NACK

Recursive-make must be avoided when possible.

For OpenThread it was not possible due to the sheer size of the OpenThread build system.

For libmetal it is possible to avoid recursive-make because the project is small and easy to build.

**Comment by MaureenHelm** [comment]
Fixed the checkpatch error and all of the warnings except for:

```
-:405: WARNING:PREFER_KERNEL_TYPES: Prefer kernel type 'u32_t' over 'uint32_t'
#405: FILE: samples/subsys/ipc/openamp/platform/resource_table.h:16:
+	uint32_t ver;

-:406: WARNING:PREFER_KERNEL_TYPES: Prefer kernel type 'u32_t' over 'uint32_t'
#406: FILE: samples/subsys/ipc/openamp/platform/resource_table.h:17:
+	uint32_t num;

-:407: WARNING:PREFER_KERNEL_TYPES: Prefer kernel type 'u32_t' over 'uint32_t'
#407: FILE: samples/subsys/ipc/openamp/platform/resource_table.h:18:
+	uint32_t reserved[2];

-:408: WARNING:PREFER_KERNEL_TYPES: Prefer kernel type 'u32_t' over 'uint32_t'
#408: FILE: samples/subsys/ipc/openamp/platform/resource_table.h:19:
+	uint32_t offset[2];
```

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**